diff --git a/CMakeLists.txt b/CMakeLists.txt
index cbc4efa..53e8c19 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -103,14 +103,14 @@ add_custom_target(circt-headers)
 set_target_properties(circt-headers PROPERTIES FOLDER "Misc")
 add_custom_target(circt-doc)
 
-# Add MLIR and LLVM headers to the include path
-include_directories(${LLVM_INCLUDE_DIRS})
-include_directories(${MLIR_INCLUDE_DIRS})
-
 # Add CIRCT files to the include path
 include_directories(${CIRCT_MAIN_INCLUDE_DIR})
 include_directories(${CIRCT_INCLUDE_DIR})
 
+# Add MLIR and LLVM headers to the include path
+include_directories(${LLVM_INCLUDE_DIRS})
+include_directories(${MLIR_INCLUDE_DIRS})
+
 #-------------------------------------------------------------------------------
 # Verilator Configuration
 #-------------------------------------------------------------------------------
