diff --git a/lib/Conversion/FIRRTLToLLHD/CMakeLists.txt b/lib/Conversion/FIRRTLToLLHD/CMakeLists.txt
index 24fcd4f..be2868e 100644
--- a/lib/Conversion/FIRRTLToLLHD/CMakeLists.txt
+++ b/lib/Conversion/FIRRTLToLLHD/CMakeLists.txt
@@ -10,4 +10,5 @@ add_mlir_conversion_library(MLIRFIRRTLToLLHD
   LINK_LIBS PUBLIC
   MLIRLLHD
   MLIRTransforms
+  MLIRFIRRTL
 )
diff --git a/lib/Dialect/FIRRTL/CMakeLists.txt b/lib/Dialect/FIRRTL/CMakeLists.txt
index f770449..1426440 100644
--- a/lib/Dialect/FIRRTL/CMakeLists.txt
+++ b/lib/Dialect/FIRRTL/CMakeLists.txt
@@ -13,6 +13,10 @@ add_mlir_dialect_library(MLIRFIRRTL
   Support
 
   LINK_LIBS PUBLIC
+  MLIRIR
+  MLIRPass
+  MLIRRTL
+  MLIRSV
    )
 
 add_dependencies(mlir-headers MLIRFIRRTLIncGen MLIRFIRRTLEnumsIncGen)
diff --git a/lib/Dialect/LLHD/IR/CMakeLists.txt b/lib/Dialect/LLHD/IR/CMakeLists.txt
index 36b271d..fb68db3 100644
--- a/lib/Dialect/LLHD/IR/CMakeLists.txt
+++ b/lib/Dialect/LLHD/IR/CMakeLists.txt
@@ -17,4 +17,5 @@ add_mlir_dialect_library(MLIRLLHD
   MLIRSideEffectInterfaces
   MLIRControlFlowInterfaces
   MLIRCallInterfaces
+  MLIRStandard
 )
diff --git a/lib/Dialect/RTL/CMakeLists.txt b/lib/Dialect/RTL/CMakeLists.txt
index 7fdb808..2f03ef5 100644
--- a/lib/Dialect/RTL/CMakeLists.txt
+++ b/lib/Dialect/RTL/CMakeLists.txt
@@ -13,6 +13,7 @@ add_mlir_dialect_library(MLIRRTL
   Support
 
   LINK_LIBS PUBLIC
+  MLIRIR
    )
 
 add_dependencies(mlir-headers MLIRRTLIncGen MLIRRTLEnumsIncGen)
diff --git a/lib/Dialect/SV/CMakeLists.txt b/lib/Dialect/SV/CMakeLists.txt
index ed4dab2..1af3e2e 100644
--- a/lib/Dialect/SV/CMakeLists.txt
+++ b/lib/Dialect/SV/CMakeLists.txt
@@ -12,6 +12,7 @@ add_mlir_dialect_library(MLIRSV
   Support
 
   LINK_LIBS PUBLIC
+  MLIRDialect
    )
 
 add_dependencies(mlir-headers MLIRSVIncGen)
diff --git a/lib/EmitVerilog/CMakeLists.txt b/lib/EmitVerilog/CMakeLists.txt
index 1853506..ff32a07 100644
--- a/lib/EmitVerilog/CMakeLists.txt
+++ b/lib/EmitVerilog/CMakeLists.txt
@@ -9,4 +9,5 @@ add_mlir_library(CIRCTEmitVerilog
   MLIRFIRRTL
   MLIRRTL
   MLIRSV
+  MLIRTranslation
   )
diff --git a/lib/FIRParser/CMakeLists.txt b/lib/FIRParser/CMakeLists.txt
index 6e973fb..c239237 100644
--- a/lib/FIRParser/CMakeLists.txt
+++ b/lib/FIRParser/CMakeLists.txt
@@ -7,4 +7,5 @@ add_mlir_library(CIRCTFIRParser
 
   LINK_LIBS PUBLIC
   MLIRFIRRTL
+  MLIRTranslation
   )
diff --git a/lib/Target/Verilog/CMakeLists.txt b/lib/Target/Verilog/CMakeLists.txt
index 86988e5..f03caef 100644
--- a/lib/Target/Verilog/CMakeLists.txt
+++ b/lib/Target/Verilog/CMakeLists.txt
@@ -6,4 +6,5 @@ add_mlir_library(MLIRLLHDTargetVerilog
 
   LINK_LIBS PUBLIC
   MLIRLLHD
+  MLIRTranslation
   )
