diff --git a/tools/protoc-gen-mypy.sh.in b/tools/protoc-gen-mypy.sh.in
old mode 100644
new mode 100755
index d589a5e9..26bc4663
--- a/tools/protoc-gen-mypy.sh.in
+++ b/tools/protoc-gen-mypy.sh.in
@@ -1,3 +1,3 @@
 #!/bin/bash
 # use PYTHON_EXECUTABLE from CMake to get the right python to execute
-@PYTHON_EXECUTABLE@ -u "$PWD"/../tools/protoc-gen-mypy.py
\ No newline at end of file
+@PYTHON_EXECUTABLE@ -u "$(dirname "$0")/protoc-gen-mypy.py"
